native/src/main/java/io/ballerina/lib/avro/deserialize/visitor/DeserializeVisitor.java

```diff
@@ -113,6 +114,9 @@ public BMap<BString, Object> visit(RecordDeserializer recordDeserializer, Generi
         if (originalType.isReadOnly()) {
             avroRecord.freezeDirect();
         }
+        if (type.getTag() == TypeTags.ANYDATA_TAG) {
+            return (BMap<BString, Object>) ValueUtils.convert(avroRecord, type);
```

Comment:
In L86 the `avroRecord` has been already defined as `BMap` (with this specific key-value types), why do we need to convert it back ?

Reply:
It's not converting back. Here the `BMap` value is converting to the `BAnydataType`. Without applying this, I'm getting a casting issue

Reply:
Ack. Better to check if we can directly create an `anydata` type using `ValueCreator` itself.

Reply:
Fixed in a11a0c9